src/presign_handler/signer.rs

```diff
@@ -110,9 +110,23 @@ impl PresignHandler {
         mac.update(query_signature.as_bytes());
 
         match mac.verify_slice(signature_hmac_key.as_slice()) {
-            Ok(_) => Ok(true),
-            Err(_) => Ok(false),
+            Ok(_) => {}
+            Err(_) => return Ok(false),
         }
+
```

Comment:
Should work for now but might be a good idea to give this invalid information to the user.